### PR TITLE
feat(provider-generator): allow removing big, cost intensive structures

### PR DESCRIPTION
**Pull request by DanielMSchmidt**
## Summary

This change allows us to keep a disallow list of provider attributes that generate an insane amount of classes (in this case over 25.000) in our bindings. I decided to not automatically detect these instances to not prematurely optimise this. For a similar reasoning and since it's not automatically detected I also skipped the implementation of attribute skipping (since it's also very uncertain that attributes will ever be the problem).

### CODE REVIEW

**Comment by team-tf-cdk** [comment]
This PR exceeds the recommended size of 1000 lines. Please make sure you are NOT addressing multiple issues with one PR. Note this PR might be rejected due to its size.

**Comment by DanielMSchmidt** [comment]
At least for go and `cdktf get` the performance didn't improve locally

**Comment by mutahhir** [comment]
> At least for go and `cdktf get` the performance didn't improve locally

The go conversion is interesting. When I was running it, it seemed to take a lot longer, but it wasn't taxing my system where I could tell that it's really computationally expensive. I think we should take a look at the JSII aspects of the Go conversion too when we get to performance.

File: packages/@cdktf/provider-generator/lib/get/__tests__/generator/fixtures/aws_quicksight_template.test.fixture.json
Comment (ansgarm) [inline packages/@cdktf/provider-generator/lib/get/__tests__/generator/fixtures/aws_quicksight_template.test.fixture.json:1]: That is a laaaarge file 😂
(Didn't read it 😄)

File: packages/@cdktf/provider-generator/lib/get/generator/models/scope.ts
Comment (ansgarm) [inline packages/@cdktf/provider-generator/lib/get/generator/models/scope.ts:69]: I feel like previous fits better 😄

```suggestion
      const prev = path[i - 1];
      return p.replace(`${prev}_`, "");
```

**Review by ansgarm — APPROVED** [review]

File: packages/@cdktf/provider-generator/lib/get/generator/models/scope.ts
Comment (jsteinich) [inline packages/@cdktf/provider-generator/lib/get/generator/models/scope.ts:55]: This seems to be used in place of `fullName`; is there a particular reason why that wasn't used?

File: packages/@cdktf/provider-generator/lib/get/generator/skipped-attributes.ts
Comment (jsteinich) [inline packages/@cdktf/provider-generator/lib/get/generator/skipped-attributes.ts:10]: Baking these in is a slippery slope. It sends a message of not supporting arbitrary resources. It also introduces breaking changes anytime the list is changed.

File: website/docs/cdktf/concepts/resources.mdx
Comment (jsteinich) [inline website/docs/cdktf/concepts/resources.mdx:925]: Might want an example of how to use them. Particularly in non-Typescript as working with `any` is quite tricky.

**Review by jsteinich — COMMENTED** [review]
This could relatively easily be extended to support some of the ideas in https://github.com/hashicorp/terraform-cdk/issues/2892. Allowing users to generate their own trimmed down providers or having a temporary escape hatch for unsupported updates is definitely appealing.
However, forcing pieces to be omitted within the generator itself seems too heavy handed, especially since the benefit doesn't appear to have been proven out.

File: packages/@cdktf/provider-generator/lib/get/generator/skipped-attributes.ts
Comment (mutahhir) [reply packages/@cdktf/provider-generator/lib/get/generator/skipped-attributes.ts:10]: Agreed on the breaking changes on each change. That's definitely inconvenient.

However, I think it's not really a slippery slope. The singular purpose of this list is to unblock provider generation for providers that have a schema design that won't be usable in any one of our supported languages. In this case, the attributes in this list prevent us from publishing any new versions of the provider. 

I agree that we may be able to come up with a more sophisticated solution to this problem, and we should. However, we don't know what that solution looks like yet and we want to allow users to be able to use newer versions of this provider through CDKTF now. To that end, I think this PR makes a pragmatic tradeoff for this situation.

**Review by mutahhir — APPROVED** [review]

File: packages/@cdktf/provider-generator/lib/get/generator/models/scope.ts
Comment (DanielMSchmidt) [reply packages/@cdktf/provider-generator/lib/get/generator/models/scope.ts:45]: To not have the baseName start with a dot

**Comment by mutahhir** [comment]
> This could relatively easily be extended to support some of the ideas in #2892. Allowing users to generate their own trimmed down providers or having a temporary escape hatch for unsupported updates is definitely appealing. However, forcing pieces to be omitted within the generator itself seems too heavy handed, especially since the benefit doesn't appear to have been proven out.

I think there's a bit of miscommunication on our part. The benefit that we were talking about in the earlier comments was regarding overall improvement in provider generation times for languages like Go, which didn't really improve (unfortunately). However, the main objective for this PR is to allow the newer releases of the AWS provider to be generated. 

I think while #2892 is an interesting idea, I think it also is limited to people who are building providers locally. Prebuilt providers have a lot of benefit for users in terms of not requiring as much compute resources for locally building providers. Thus, unless certain situations require it, prebuilt providers would be the more popular choice.

File: website/docs/cdktf/concepts/resources.mdx
Comment (DanielMSchmidt) [reply website/docs/cdktf/concepts/resources.mdx:925]: Yeah good point, I'll add some in a separate PR 👍

**Pull request merged**
